pkg/apis/intevents/v1/topic_validation.go

```diff
@@ -54,16 +56,14 @@ func (current *Topic) CheckImmutableFields(ctx context.Context, original *Topic)
 		return nil
 	}
 
-	var errs *apis.FieldError
-
-	// Topic is immutable.
-	if original.Spec.Topic != current.Spec.Topic {
-		errs = errs.Also(
-			&apis.FieldError{
-				Message: "Immutable field changed",
-				Paths:   []string{"spec", "topic"},
-				Details: fmt.Sprintf("was %q, now %q", original.Spec.Topic, current.Spec.Topic),
-			})
+	// Modification of Topic, Secret, ServiceAccountName and Project are not allowed. Everything else is mutable.
+	if diff := cmp.Diff(original.Spec, current.Spec,
+		cmpopts.IgnoreFields(TopicSpec{}, "PropagationPolicy", "EnablePublisher")); diff != "" {
```

Comment:
@nachocano, could you please check the logic is ok here? You mentioned that all fields that are immutable in pullsubscription should also be immutable here, does that mean `PropagationPolicy` and `EnablePublisher` are mutable then?

Reply:
my worry of letting them be mutable is that if we change them, then our reconciler might not properly reconcile those changes. I think for these two it should be ok. We can leave them mutable for now.
Do you mind creating a new issue to check immutability constraints in all of our resources, and whether we are doing it right. We can then tackle that in the following sprint.

Reply:
Sure, issue created #1448